ssh-key/src/private.rs

```diff
@@ -344,50 +344,44 @@ impl PrivateKey {
         SshSig::sign(self, namespace, hash_alg, msg)
     }
 
-    /// Read private key from an OpenSSH-formatted PEM file.
+    /// Read private key from an OpenSSH-formatted PEM source.
     #[cfg(feature = "std")]
-    pub fn read_openssh<R: Read>(reader: &mut R) -> Result<Self> {
+    pub fn read_openssh(reader: &mut impl Read) -> Result<Self> {
         let pem = Zeroizing::new(io::read_to_string(reader)?);
         Self::from_openssh(&*pem)
     }
 
     /// Read private key from an OpenSSH-formatted PEM file.
     #[cfg(feature = "std")]
-    pub fn read_openssh_file<P: AsRef<Path>>(path: P) -> Result<Self> {
+    pub fn read_openssh_file(path: impl AsRef<Path>) -> Result<Self> {
         // TODO(tarcieri): verify file permissions match `UNIX_FILE_PERMISSIONS`
-        let pem = Zeroizing::new(fs::read_to_string(path)?);
-        Self::from_openssh(&*pem)
+        let mut file = File::open(path)?;
+        Self::read_openssh(&mut file)
     }
 
     /// Write private key as an OpenSSH-formatted PEM file.
     #[cfg(feature = "std")]
-    pub fn write_openssh<W: Write>(&self, writer: &mut W, line_ending: LineEnding) -> Result<()> {
+    pub fn write_openssh(&self, writer: &mut impl Write, line_ending: LineEnding) -> Result<()> {
         let pem = self.to_openssh(line_ending)?;
         writer.write_all(pem.as_bytes())?;
         Ok(())
     }
 
     /// Write private key as an OpenSSH-formatted PEM file.
     #[cfg(feature = "std")]
-    pub fn write_openssh_file<P: AsRef<Path>>(
+    pub fn write_openssh_file(
         &self,
-        path: P,
+        path: impl AsRef<Path>,
         line_ending: LineEnding,
     ) -> Result<()> {
-        let pem = self.to_openssh(line_ending)?;
+        let mut options = File::options();
```

Comment:
Can you put all of the default options here? I guess you're doing this to avoid `unused_mut` on Windows:

```rust
let mut opyiond = File::options().write(true).create(true).truncate(true).open(path)?;
```

Reply:
Yup. The only reason I did that was to avoid clippy's `unused_mut` check.
I can use a block expression for `options` so that the `.open(path)` line is cleaner.